lightning/src/chain/package.rs

```diff
@@ -1659,4 +1685,53 @@ mod tests {
 			}
 		}
 	}
+
+	struct TestFeeEstimator {
+		sat_per_kw: u32,
+	}
+
+	impl FeeEstimator for TestFeeEstimator {
+		fn get_est_sat_per_1000_weight(&self, _: ConfirmationTarget) -> u32 {
+			self.sat_per_kw
+		}
+	}
+
+	#[test]
+	fn test_feerate_bump() {
+		let sat_per_kw = FEERATE_FLOOR_SATS_PER_KW;
+		let test_fee_estimator = &TestFeeEstimator { sat_per_kw };
+		let fee_estimator = LowerBoundedFeeEstimator::new(test_fee_estimator);
+		let fee_rate_strategy = FeerateStrategy::ForceBump;
+		let confirmation_target = ConfirmationTarget::UrgentOnChainSweep;
+		let logger = TestLogger::new();
+
+		{
+			let predicted_weight_units = 1000;
+			let input_satoshis = 861;
+
+			let bumped_fee_rate = feerate_bump(predicted_weight_units, input_satoshis, 546, 253, &fee_rate_strategy, confirmation_target, &fee_estimator, &logger);
+			assert!(bumped_fee_rate.is_none());
+			logger.assert_log_regex("lightning::chain::package", regex::Regex::new(r"Can't new-estimation bump new claiming tx, output amount 545 would end up below dust threshold 546").unwrap(), 1);
```

Comment:
This doesn't seem right...

- Weight: 1000
- Input amount: 861
- Previous feerate: 253
- `new_fee = max(1.25 * 253, 253 + 253) = 506`
- `remaining_output_amount = 861 - 506 = 355`

Yet test test expects output amount of 545.

What am I missing?

Reply:
Got it, thanks for explaining.  By the way, it looks like that early termination path can be removed since the dust check is repeated here as well.

Reply:
My intuition is that during debugging most of the path can be inferred by the amount of the fee bump (e.g., is it 25% or the incremental relay fee), especially for a short, fairly simple function like this one.  In cases where something unexpected could happen, that's when logging is more helpful.

For instance, I'm not sure it's super helpful to have logs like this:

```
... Initiating fee rate bump from X to Y.
... Attempting forced 25% fee rate bump from X to (X*1.25)
... Naive fee bump of (X*.25) does not meet min relay fee of 253
... Fee rate bumped by 253 from X to (X+253)
```

when something like this communicates almost as much:

```
... Fee rate bumped from X to X+253
```